### DISCUSSION TITLE
New feature: convert Steam Community modals to plain links

### DISCUSSION THREAD

**Discussion by denilsonsa** [IsThereAnyDeal/AugmentedSteam#1924] [Ideas]
When viewing Steam Community items, each "card" is a `<div>` with some JavaScript to open it as an in-page modal dialog. The UX isn't very good, as it is essentially reinventing an hyperlink but with fewer features.

So, I'm proposing AugmentedSteam to include a feature to convert them into plain HTML `<a href="…">` links. And, even better, I'm providing you with the code to achieve that, as I've implemented it myself as a user-script:

* <https://greasyfork.org/en/scripts/498305-steam-community-modals-as-links>
* <https://gist.github.com/denilsonsa/c18dd78cba66832720176728f33f0c01>

Feel free to use that code and include it into the extension.

The `debouncedMutationObserver` is something I wrote for [another user-js](https://greasyfork.org/en/scripts/478401-bundle-helper-reborn), and I shamelessly and blindly reused. Feel free to get rid of it and just add the main code (about a dozen lines) into Augmented Steam's own Mutation Observer (assuming there is already one).

**Comment by candela97** [comment]
Thanks for the suggestion! Sounds like an ok feature to add with an option.

We can probably patch the handler in modalContent.js instead of using a mutation observer 🤔

**Reply by denilsonsa** [reply]
> We can probably patch the handler in modalContent.js instead of using a mutation observer 🤔

Probably not, I believe.

It looks like `modalContent.js` attaches a `click` event listener to the document, and then filters events based on a selector. It's been years since I last used jQuery, so I'm not sure.

Instead, my suggestion involves actually replacing `<div>` elements with native `<a href="…">` elements, which requires modifying the DOM whenever new elements are created (i.e. when scrolling to the bottom). I don't see how `modalContent.js` could help us in that. I'd be happy to be shown another solution, I just can't see any other solution right now.
